langlib/lang.table/src/main/ballerina/table.bal

```diff
@@ -86,6 +115,27 @@ public isolated function get(table<MapType> key<KeyType> t, KeyType k) returns M
 # otherwise, it will be added as the last member.
 # It panics if parameter `val` is inconsistent with the inherent type of parameter `t`.
 #
+# ```ballerina
+# table<record {|readonly int id; string name;|}> key(id) students = table [
+#     {id: 1, name: "Jo"},
+#     {id: 2, name: "Sam"}
+# ];
+# 
+# students.put({id: 3, name: "Pat"});
+# students ⇒ [{"id":1,"name":"Jo"},{"id":2,"name":"Sam"},{"id":3,"name":"Pat"}]
+# 
+# students.put({id: 1, name: "Kane"});
+# students ⇒ [{"id":1,"name":"Kane"},{"id":2,"name":"Sam"},{"id":3,"name":"Pat"}]
+# 
+# table[
+#      {id: 1, name: "Jo"},
+#      {id: 2, name: "Sam"}
+#  ].put({id: 1, name: "Pat"}) ⇒ [{"id":1,"name":"Jo"},{"id":2,"name":"Sam"},{"id":1,"name":"Pat"}]
```

Comment:
```suggestion
```suggestion
# var tbl = table [
#         {id: 1, name: "Jo"},
#         {id: 2, name: "Sam"}
#     ];
# tbl.put({id: 1, name: "Pat"});
# tbl ⇒ [{"id":1,"name":"Jo"},{"id":2,"name":"Sam"},{"id":1,"name":"Pat"}]
```

Reply:
We'll have to change `add` as well like above. This is because when we do 
```ballerina
table [{...}{...}].put({...}) ⇒ [{...},{...},{...}]
```
the output we give after ⇒ is not correct which can be misleading IMO. The actual output of this nested expression is `never`.
You can try it out by enclosing the whole thing inside `println`
```ballerina
     io:println(table [
        {id: 1, name: "Jo"},
        {id: 2, name: "Sam"}
    ].put({id: 1, name:  "Pat"}));
```